### PR TITLE
fix(Store): bootstrap store with partial initial state

### LINKED ISSUE #906: Store: Handling of @ngrx/store/update-reducers Action on Startup leads to strange behaviour for initialStates
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
</code></pre>

## What is the current behavior?
I configure a object as an initialState for the store. Unfortunately some values are "overridden" on start-up of the store/app.

## Expected behavior:
All properties from the initial-state should remain untouched by any internal ngrx-action.

## Minimal reproduction of the problem with instructions:
I think my explanation was i bit poor sofar (sorry it's hard to explain this). I think an example will show the problem:

https://stackblitz.com/edit/angular-6wbjq4

In this example everything is fine. The initial state is produced by the initState() function in app/app.state.ts. 
screenshot 2018-03-12 18 02 05

*BUT*:

Remove the key 'featureB' from this init state will "break" this behaviour. 

screenshot 2018-03-12 18 03 42

It seems to be caused by the combination function: https://github.com/ngrx/platform/blob/9b02f79f1fc3b5e4d85bf0516108f93eed655126/modules/store/src/utils.ts#L29-L44

The function expects all possible state properties to be setted. If a property is not (in initialState) set the 'hasChanged' turns true and returns the nextState. This is some kind of an intermediated state (at least while the  ' @ngrx/store/update-reducers'-Action is fired).

## Version of affected browser(s),operating system(s), npm, node and ngrx:
@ngrx/store v 5.1 + 5.2

## Other information:

This is not directly linked to the problem but another finding that confused me.
After '@ngrx/store/update-reducers' is fired all reducers are applied, even if they were already applied previously:

screenshot 2018-03-12 18 09 37

### PR DESCRIPTION
**Pull request by tdeschryver**
This changes the way features are added and removed.
Instead of adding/removing feature by feature, this adds/removes them in all together in batch.

With consequence that will only dispatch `@ngrx/store/update-reducers` once, if this isn't what we want we can still dispatch every feature one by one.

The "original" `addFeature`, `removeFeature`, `addReducer`, `removeReducer` methods will loop through the corresponding `addFeatures`, `removeFeatures`, `addReducers`, `removeReducers` methods.

Will it be beneficial to also add `addReducers` and `removeReducers` in `store.ts`?

This closes #906 and closes #909

### CODE REVIEW

File: modules/store/src/utils.ts [hunk 1]
```diff
@@ -1,117 +1,117 @@
-import {
```
Comment (tdeschryver) [inline modules/store/src/utils.ts]: I don't know what happened here? 😅

Comment (brandonroberts) [reply modules/store/src/utils.ts]: Revert these changes

File: modules/store/src/reducer_manager.ts [hunk 2]
```diff
@@ -1,89 +1,118 @@
...
+    featureKeys.forEach(key => {
+      this.reducers = omit(this.reducers, key) /*TODO(#823)*/ as any;
+    });
+    this.updateReducers(featureKeys);
+  }
+
+  private updateReducers(keys: string[]) {
+    this.next(this.reducerFactory(this.reducers, this.initialState));
+
+    if (keys.length === 1) {
```
Comment (brandonroberts) [inline modules/store/src/reducer_manager.ts]: This would be a breaking change. Currently there is one dispatch per reducer update.

**Comment by brandonroberts** [comment]
Don't know what's going on with the whitespace changes. Update the commit message also

File: modules/store/src/reducer_manager.ts (hunk 2 above)
Comment (tdeschryver) [reply modules/store/src/reducer_manager.ts]: To be sure, we want to keep dispatching every update?

Comment (brandonroberts) [reply modules/store/src/reducer_manager.ts]: Yes, dispatch individual updates for each key added. In V7, we'll change it dispatch one action.

**Pull request merged**